### ISSUE TITLE
Cinnamon crash when opening PDF from thunderbird

### ISSUE THREAD

**Issue by desbma** [linuxmint/Cinnamon#8085]
* Cinnamon 4.0.3
* Arch Linux 64bits up to date
* Nvidia GPU with proprietary drivers

Cinnamon crashes when opening a PDF attachment from Thunderbird.
Opening the same PDF outside of Thunderbird works fine.

Stack trace of all threads as reported by systemd:
```
nov. 29 22:45:12 xxx systemd-coredump[4959]: Process 2425 (cinnamon) of user 1000 dumped core.
                                              
                                              Stack trace of thread 2425:
                                              #0  0x00007f1f94a43f20 cairo_region_num_rectangles (libcairo.so.2)
                                              #1  0x00007f1f9654bb14 n/a (libmuffin.so.0)
                                              #2  0x00007f1f96550c94 n/a (libmuffin.so.0)
                                              #3  0x00007f1f96550d77 n/a (libmuffin.so.0)
                                              #4  0x00007f1f96551ae0 n/a (libmuffin.so.0)
                                              #5  0x00007f1f9345d6d0 ffi_call_unix64 (libffi.so.6)
                                              #6  0x00007f1f9345d0a0 ffi_call (libffi.so.6)
                                              #7  0x00007f1f95c78c7c n/a (libcjs.so.0)
                                              #8  0x00007f1f95c7a4c7 n/a (libcjs.so.0)
                                              #9  0x00007f1f91de9c48 n/a (libmozjs-52.so.0)
                                              #10 0x00007f1f91dfa460 n/a (libmozjs-52.so.0)
                                              #11 0x00007f1f91de91ae n/a (libmozjs-52.so.0)
                                              #12 0x00007f1f91de9ada n/a (libmozjs-52.so.0)
                                              #13 0x00007f1f91de9f78 n/a (libmozjs-52.so.0)
                                              #14 0x00007f1f91b3ef35 n/a (libmozjs-52.so.0)
                                              #15 0x00007f1f91de9baf n/a (libmozjs-52.so.0)
                                              #16 0x00007f1f91dfa460 n/a (libmozjs-52.so.0)
                                              #17 0x00007f1f91de91ae n/a (libmozjs-52.so.0)
                                              #18 0x00007f1f91de9ada n/a (libmozjs-52.so.0)
                                              #19 0x00007f1f91de9f78 n/a (libmozjs-52.so.0)
                                              #20 0x00007f1f91b3ef35 n/a (libmozjs-52.so.0)
                                              #21 0x00007f1f91de9baf n/a (libmozjs-52.so.0)
                                              #22 0x00007f1f91e330ec n/a (libmozjs-52.so.0)
                                              #23 0x00003def944f56d6 n/a (n/a)
                                              #24 0x000056039fd180f0 n/a (n/a)
                                              #25 0x00003def944eb888 n/a (n/a)
                                              #26 0x00007f1f91e2b831 n/a (libmozjs-52.so.0)
                                              #27 0x00007f1f91e2bb8b n/a (libmozjs-52.so.0)
                                              #28 0x00007f1f91de940b n/a (libmozjs-52.so.0)
                                              #29 0x00007f1f91de9ada n/a (libmozjs-52.so.0)
                                              #30 0x00007f1f91e330ec n/a (libmozjs-52.so.0)
                                              #31 0x00003def944f56d6 n/a (n/a)
                                              #32 0x00005603a1d5e750 n/a (n/a)
                                              #33 0x00003def9454d4a1 n/a (n/a)
                                              #34 0x00005603a10e7698 n/a (n/a)
                                              #35 0x00003def9454d4a1 n/a (n/a)
                                              #36 0x000056039fec7648 n/a (n/a)
                                              #37 0x00003def9461171a n/a (n/a)
                                              #38 0x00005603a133d838 n/a (n/a)
                                              #39 0x00003def9454d4a1 n/a (n/a)
                                              #40 0x00005603a0619bd0 n/a (n/a)
                                              #41 0x00003def9454d4a1 n/a (n/a)
                                              #42 0x00005603a1d86d58 n/a (n/a)
                                              #43 0x00003def9454d4a1 n/a (n/a)
                                              #44 0x00005603a0f772f8 n/a (n/a)
                                              #45 0x00003def944eb888 n/a (n/a)
                                              #46 0x00007f1f91e2b831 n/a (libmozjs-52.so.0)
                                              #47 0x00007f1f91e2bb8b n/a (libmozjs-52.so.0)
                                              #48 0x00007f1f91de940b n/a (libmozjs-52.so.0)
                                              #49 0x00007f1f91de9ada n/a (libmozjs-52.so.0)
                                              #50 0x00007f1f91de9f78 n/a (libmozjs-52.so.0)
                                              #51 0x00007f1f91b08779 _Z20JS_CallFunctionValueP9JSContextN2JS6HandleIP8JSObjectEENS2_INS1_5ValueEEERKNS1_16HandleValueArrayENS1_13MutableHandleIS6_EE (libmozjs-52.so.0)
                                              #52 0x00007f1f95c9cf22 gjs_call_function_value (libcjs.so.0)
                                              #53 0x00007f1f95c73c2d gjs_closure_invoke (libcjs.so.0)
                                              #54 0x00007f1f95c90e68 n/a (libcjs.so.0)
                                              #55 0x00007f1f960873d5 g_closure_invoke (libgobject-2.0.so.0)
                                              #56 0x00007f1f96074195 n/a (libgobject-2.0.so.0)
                                              #57 0x00007f1f9607801e g_signal_emit_valist (libgobject-2.0.so.0)
                                              #58 0x00007f1f96078a80 g_signal_emit (libgobject-2.0.so.0)
                                              #59 0x00007f1f96479684 n/a (libmuffin-clutter-0.so)
                                              #60 0x00007f1f964500e1 n/a (libmuffin-clutter-0.so)
                                              #61 0x00007f1f95f9f3cf g_main_context_dispatch (libglib-2.0.so.0)
                                              #62 0x00007f1f95fa0f89 n/a (libglib-2.0.so.0)
                                              #63 0x00007f1f95fa1f62 g_main_loop_run (libglib-2.0.so.0)
                                              
                                              Stack trace of thread 2449:
                                              #0  0x00007f1f95eecafc pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f1f918eaaed n/a (libmozjs-52.so.0)
                                              #2  0x00007f1f91dffdda n/a (libmozjs-52.so.0)
                                              #3  0x00007f1f91e01201 n/a (libmozjs-52.so.0)
                                              #4  0x00007f1f91dc7aba n/a (libmozjs-52.so.0)
                                              #5  0x00007f1f95ee6a9d start_thread (libpthread.so.0)
                                              #6  0x00007f1f95e16b23 __clone (libc.so.6)
                                              
                                              Stack trace of thread 2453:
                                              #0  0x00007f1f95eecafc pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f1f918eaaed n/a (libmozjs-52.so.0)
                                              #2  0x00007f1f91dffdda n/a (libmozjs-52.so.0)
                                              #3  0x00007f1f91e01201 n/a (libmozjs-52.so.0)
                                              #4  0x00007f1f91dc7aba n/a (libmozjs-52.so.0)
                                              #5  0x00007f1f95ee6a9d start_thread (libpthread.so.0)
                                              #6  0x00007f1f95e16b23 __clone (libc.so.6)
                                              
                                              Stack trace of thread 2452:
                                              #0  0x00007f1f95eecafc pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f1f918eaaed n/a (libmozjs-52.so.0)
                                              #2  0x00007f1f91dffdda n/a (libmozjs-52.so.0)
                                              #3  0x00007f1f91e01201 n/a (libmozjs-52.so.0)
                                              #4  0x00007f1f91dc7aba n/a (libmozjs-52.so.0)
                                              #5  0x00007f1f95ee6a9d start_thread (libpthread.so.0)
                                              #6  0x00007f1f95e16b23 __clone (libc.so.6)
                                              
                                              Stack trace of thread 2451:
                                              #0  0x00007f1f95eecafc pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f1f918eaaed n/a (libmozjs-52.so.0)
                                              #2  0x00007f1f91dffdda n/a (libmozjs-52.so.0)
                                              #3  0x00007f1f91e01201 n/a (libmozjs-52.so.0)
                                              #4  0x00007f1f91dc7aba n/a (libmozjs-52.so.0)
                                              #5  0x00007f1f95ee6a9d start_thread (libpthread.so.0)
                                              #6  0x00007f1f95e16b23 __clone (libc.so.6)
                                              
                                              Stack trace of thread 2448:
                                              #0  0x00007f1f95eecafc pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f1f918eaaed n/a (libmozjs-52.so.0)
                                              #2  0x00007f1f91dffdda n/a (libmozjs-52.so.0)
                                              #3  0x00007f1f91e01201 n/a (libmozjs-52.so.0)
                                              #4  0x00007f1f91dc7aba n/a (libmozjs-52.so.0)
                                              #5  0x00007f1f95ee6a9d start_thread (libpthread.so.0)
                                              #6  0x00007f1f95e16b23 __clone (libc.so.6)
                                              
                                              Stack trace of thread 2445:
                                              #0  0x00007f1f95eecafc pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f1f918eaaed n/a (libmozjs-52.so.0)
                                              #2  0x00007f1f91dffdda n/a (libmozjs-52.so.0)
                                              #3  0x00007f1f91e01201 n/a (libmozjs-52.so.0)
                                              #4  0x00007f1f91dc7aba n/a (libmozjs-52.so.0)
                                              #5  0x00007f1f95ee6a9d start_thread (libpthread.so.0)
                                              #6  0x00007f1f95e16b23 __clone (libc.so.6)
                                              
                                              Stack trace of thread 2439:
                                              #0  0x00007f1f95e0bc21 __poll (libc.so.6)
                                              #1  0x00007f1f95fa0ee0 n/a (libglib-2.0.so.0)
                                              #2  0x00007f1f95fa0fce g_main_context_iteration (libglib-2.0.so.0)
                                              #3  0x00007f1f95fa1022 n/a (libglib-2.0.so.0)
                                              #4  0x00007f1f95f6a3eb n/a (libglib-2.0.so.0)
                                              #5  0x00007f1f95ee6a9d start_thread (libpthread.so.0)
                                              #6  0x00007f1f95e16b23 __clone (libc.so.6)
                                              
                                              Stack trace of thread 2455:
                                              #0  0x00007f1f95eecafc pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f1f918eaaed n/a (libmozjs-52.so.0)
                                              #2  0x00007f1f91dffdda n/a (libmozjs-52.so.0)
                                              #3  0x00007f1f91e01201 n/a (libmozjs-52.so.0)
                                              #4  0x00007f1f91dc7aba n/a (libmozjs-52.so.0)
                                              #5  0x00007f1f95ee6a9d start_thread (libpthread.so.0)
                                              #6  0x00007f1f95e16b23 __clone (libc.so.6)
                                              
                                              Stack trace of thread 2442:
                                              #0  0x00007f1f95eecafc pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f1f918eaaed n/a (libmozjs-52.so.0)
                                              #2  0x00007f1f91dffdda n/a (libmozjs-52.so.0)
                                              #3  0x00007f1f91e01201 n/a (libmozjs-52.so.0)
                                              #4  0x00007f1f91dc7aba n/a (libmozjs-52.so.0)
                                              #5  0x00007f1f95ee6a9d start_thread (libpthread.so.0)
                                              #6  0x00007f1f95e16b23 __clone (libc.so.6)
                                              
                                              Stack trace of thread 2450:
                                              #0  0x00007f1f95eecafc pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f1f918eaaed n/a (libmozjs-52.so.0)
                                              #2  0x00007f1f91dffdda n/a (libmozjs-52.so.0)
                                              #3  0x00007f1f91e01201 n/a (libmozjs-52.so.0)
                                              #4  0x00007f1f91dc7aba n/a (libmozjs-52.so.0)
                                              #5  0x00007f1f95ee6a9d start_thread (libpthread.so.0)
                                              #6  0x00007f1f95e16b23 __clone (libc.so.6)
                                              
                                              Stack trace of thread 2444:
                                              #0  0x00007f1f95eecafc pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f1f918eaaed n/a (libmozjs-52.so.0)
                                              #2  0x00007f1f91dffdda n/a (libmozjs-52.so.0)
                                              #3  0x00007f1f91e01201 n/a (libmozjs-52.so.0)
                                              #4  0x00007f1f91dc7aba n/a (libmozjs-52.so.0)
                                              #5  0x00007f1f95ee6a9d start_thread (libpthread.so.0)
                                              #6  0x00007f1f95e16b23 __clone (libc.so.6)
                                              
                                              Stack trace of thread 2457:
                                              #0  0x00007f1f95eecafc pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f1f918eaaed n/a (libmozjs-52.so.0)
                                              #2  0x00007f1f91dffdda n/a (libmozjs-52.so.0)
                                              #3  0x00007f1f91e01201 n/a (libmozjs-52.so.0)
                                              #4  0x00007f1f91dc7aba n/a (libmozjs-52.so.0)
                                              #5  0x00007f1f95ee6a9d start_thread (libpthread.so.0)
                                              #6  0x00007f1f95e16b23 __clone (libc.so.6)
                                              
                                              Stack trace of thread 2443:
                                              #0  0x00007f1f95eecafc pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f1f918eaaed n/a (libmozjs-52.so.0)
                                              #2  0x00007f1f91dffdda n/a (libmozjs-52.so.0)
                                              #3  0x00007f1f91e01201 n/a (libmozjs-52.so.0)
                                              #4  0x00007f1f91dc7aba n/a (libmozjs-52.so.0)
                                              #5  0x00007f1f95ee6a9d start_thread (libpthread.so.0)
                                              #6  0x00007f1f95e16b23 __clone (libc.so.6)
                                              
                                              Stack trace of thread 2441:
                                              #0  0x00007f1f95e0bc21 __poll (libc.so.6)
                                              #1  0x00007f1f95fa0ee0 n/a (libglib-2.0.so.0)
                                              #2  0x00007f1f95fa0fce g_main_context_iteration (libglib-2.0.so.0)
                                              #3  0x00007f1f8c534bde n/a (libdconfsettings.so)
                                              #4  0x00007f1f95f6a3eb n/a (libglib-2.0.so.0)
                                              #5  0x00007f1f95ee6a9d start_thread (libpthread.so.0)
                                              #6  0x00007f1f95e16b23 __clone (libc.so.6)
                                              
                                              Stack trace of thread 2446:
                                              #0  0x00007f1f95eecafc pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f1f918eaaed n/a (libmozjs-52.so.0)
                                              #2  0x00007f1f91dffdda n/a (libmozjs-52.so.0)
                                              #3  0x00007f1f91e01201 n/a (libmozjs-52.so.0)
                                              #4  0x00007f1f91dc7aba n/a (libmozjs-52.so.0)
                                              #5  0x00007f1f95ee6a9d start_thread (libpthread.so.0)
                                              #6  0x00007f1f95e16b23 __clone (libc.so.6)
                                              
                                              Stack trace of thread 2447:
                                              #0  0x00007f1f95eecafc pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f1f918eaaed n/a (libmozjs-52.so.0)
                                              #2  0x00007f1f91dffdda n/a (libmozjs-52.so.0)
                                              #3  0x00007f1f91e01201 n/a (libmozjs-52.so.0)
                                              #4  0x00007f1f91dc7aba n/a (libmozjs-52.so.0)
                                              #5  0x00007f1f95ee6a9d start_thread (libpthread.so.0)
                                              #6  0x00007f1f95e16b23 __clone (libc.so.6)
                                              
                                              Stack trace of thread 2456:
                                              #0  0x00007f1f95eecafc pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f1f918eaaed n/a (libmozjs-52.so.0)
                                              #2  0x00007f1f91dffdda n/a (libmozjs-52.so.0)
                                              #3  0x00007f1f91e01201 n/a (libmozjs-52.so.0)
                                              #4  0x00007f1f91dc7aba n/a (libmozjs-52.so.0)
                                              #5  0x00007f1f95ee6a9d start_thread (libpthread.so.0)
                                              #6  0x00007f1f95e16b23 __clone (libc.so.6)
                                              
                                              Stack trace of thread 2440:
                                              #0  0x00007f1f95e0bc21 __poll (libc.so.6)
                                              #1  0x00007f1f95fa0ee0 n/a (libglib-2.0.so.0)
                                              #2  0x00007f1f95fa1f62 g_main_loop_run (libglib-2.0.so.0)
                                              #3  0x00007f1f96109c28 n/a (libgio-2.0.so.0)
                                              #4  0x00007f1f95f6a3eb n/a (libglib-2.0.so.0)
                                              #5  0x00007f1f95ee6a9d start_thread (libpthread.so.0)
                                              #6  0x00007f1f95e16b23 __clone (libc.so.6)
                                              
                                              Stack trace of thread 2454:
                                              #0  0x00007f1f95eecafc pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                              #1  0x00007f1f918eaaed n/a (libmozjs-52.so.0)
                                              #2  0x00007f1f91dffdda n/a (libmozjs-52.so.0)
                                              #3  0x00007f1f91e01201 n/a (libmozjs-52.so.0)
                                              #4  0x00007f1f91dc7aba n/a (libmozjs-52.so.0)
                                              #5  0x00007f1f95ee6a9d start_thread (libpthread.so.0)
                                              #6  0x00007f1f95e16b23 __clone (libc.so.6)
-- Subject: Le processus 2425 (cinnamon) a généré un fichier « core »
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- Documentation: man:core(5)
-- 
-- Le processus 2425 (cinnamon) a planté et généré un fichier « core ».
-- 
-- Cela indique généralement une erreur de programmation dans le programme
-- incriminé, et cela devrait être notifié à son concepteur comme un défaut (bug).
```

If that helps, I can also upload the core dump.

This is a regression of 4.x and it worked with Cinnamon 3.x.

**Comment by leigh123linux** [comment]
The incomplete backtrace  is for muffin

**Comment by leigh123linux** [comment]
Thunderbird uses an external app to open pdf, xreader opens attachments fine here.

**Comment by desbma** [comment]
I'm using Evince.
I just tried Xreader (2.0.1), and I get the exact same crash.

**Comment by desbma** [comment]
I can confirm this is fixed with Cinnamon 4.0.5.
